File: barras_agrupadas.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# diamonds = pd.read_csv("caminho/para/seu/arquivo.csv")

# Carrega o conjunto de dados 'diamonds' fornecido pela seaborn
diamonds = sns.load_dataset('diamonds')

# Criar o gráfico de barras agrupadas
plt.figure(figsize=(10, 6))
# hue é a variável que agrupa os dados
sns.countplot(data=diamonds, x='cut', hue='color', palette='pastel', dodge=True)
plt.title('Qualidade do corte de diamante de acordo com a cor')
plt.xlabel('Corte')
plt.ylabel('Frequência')
plt.legend(title='Cor')
plt.show()

File: barras_agrupadas.R
library(dplyr)
library(ggplot2)
library(hrbrthemes)
library(readr)

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# diamonds <- read.csv("caminho/para/seu/arquivo.csv", header = TRUE)

# Criar o gráfico de barras agrupadas
  diamonds %>%
    # fill é a variável que agrupa os dados
    ggplot(aes(x = cut, fill = color)) +
    geom_bar(position = "dodge") +
    ggtitle("Qualidade do corte de diamante de acordo com a cor") +
    xlab("corte") +
    ylab("frequência") +
    labs(fill = "cor") +
    theme_ipsum() +
    theme(axis.text.x = element_text(size = 14),
          axis.text.y = element_text(size = 14),
          axis.title.x = element_text(size = 14),
          axis.title.y = element_text(size = 14),
          legend.title = element_text(size = 14),
          legend.text = element_text(size = 14))
